core/cloudflow-akka/src/main/scala/cloudflow/akkastream/persistence/AkkaPersistenceConfigurator.scala

```diff
@@ -0,0 +1,215 @@
+          .parseMap(cMap.asJava)
+          .withFallback(ConfigFactory.parseResourcesAnySyntax(baseCassandraConfig))
+          .withFallback(baseConfig)
+
+      case jdbcConfiguration: JDBCConfiguration => // It is JDBC config
+        log.info(s"Building JDBC configuration from $jdbcConfiguration")
+        val jdbcMap = Map(
+          JDBCPROFILE  -> jdbcConfiguration.profile,
+          JDBCURL      -> jdbcConfiguration.url,
+          JDBCUSER     -> jdbcConfiguration.user,
+          JDBCPASSWORD -> jdbcConfiguration.passwrd,
+          JDBCDRIVER   -> jdbcConfiguration.driver
+        )
+        ConfigFactory
+          .parseMap(jdbcMap.asJava)
+          .withFallback(ConfigFactory.parseResourcesAnySyntax(baseJDBCConfig))
+          .withFallback(baseConfig)
+      case _ => baseConfig
+    }
+
+  // IN the case of local run persistence configuration is part of the configuration passed to streamlet
+  def getPersistenceParamsLocal(config: Config, streamletName: String): Option[DBConfiguration] = {
+
+    val STREAMLETPERSISTENCE = s"cloudflow.streamlets.${streamletName}.config.akka.persistence"
+    // Try streamlet context first
+    config.as[Option[Config]](STREAMLETPERSISTENCE) match {
+      case Some(persistence) => // we have definition
+        internalPopulatePersistence(persistence)
+      case _ => // try runtime
+        config.as[Option[Config]](RUNTIMEPERSISTENCE) match {
+          case Some(persistence) => // we have definition
+            internalPopulatePersistence(persistence)
+          case _ => None
+        }
+    }
+  }
+
+  // In the case of running on cluster the easiest way to get configuration
+  // is from secret
+  def getPersistenceParamsCluster(): Option[DBConfiguration] = {
+
+    val secretPath = Paths.get(SECRETLOCATION)
+
+    if (Files.exists(secretPath)) {
+      log.info(s"Loading persistence configuration from $secretPath")
+      // secret takes precedence, since it contains config.
+      val runtimeConfig = ConfigFactory.parseFile(secretPath.toFile)
+      runtimeConfig.as[Option[Config]](SECRETPERSISTENCE) match {
+        case Some(persistence) => // We have persistence
+          internalPopulatePersistence(persistence)
+        case _ => None
+      }
+    } else {
+      None
+    }
+  }
+
+  private def internalPopulatePersistence(config: Config): Option[DBConfiguration] = {
+    val dbType = config.as[Option[String]](DBTYPE).getOrElse("unknown")
+    dbType match {
+      case db if (db.toLowerCase == "cassandra") => // Cassandra
+        val host = config.as[Option[String]](HOST).getOrElse("127.0.0.1")
+        val port = config.as[Option[String]](PORT).getOrElse("9042")
+        val configuration = CassandraConfiguration(
+          s"$host:$port",
+          config.as[Option[String]](DATABASE).getOrElse("akkapersistence"),
+          config.as[Option[String]](USER).getOrElse("cassandra"),
+          config.as[Option[String]](PASSWORD).getOrElse("cassandra")
+        )
```

Comment:
It is very bad practice to have config default values in the code.
(I acknowledge that Cloudflow's config resolving is challenging.)

Reply:
The config resolving has changed considerably @ennru it is now all done in the cli (which was not a real option before when it was written in Golang).